### PR TITLE
Implement filter tests for complex signals

### PR DESCRIPTION
**Pull request by tluebeck**
unit testing for filtering complex signals

### CODE REVIEW

**Comment by f-brinkmann** [comment]
We might get away with fewer tests if we test only if the three filter classes FilterFIR, FilterIIR, and FilterSOS can handle complex input. If this is the case, all pyfar filter functions will be good, as far as I understand.

File: tests/test_filter_classes.py [hunk 1]
```diff
@@ -249,6 +273,24 @@ def test_filter_sos_process(impulse):
     npt.assert_allclose(res.time, np.atleast_2d(desired))
 
 
+def test_filter_sos_process_complex(impulse_complex):
+    sos = np.array([[1, 1/2, 0, 1, 0, 0]])
+    filt = fo.FilterSOS(sos, impulse_complex.sampling_rate)
+    coeff = np.array([[1, 1/2, 0], [1, 0, 0]])
```
Comment (f-brinkmann) [inline tests/test_filter_classes.py:279]: Why are you initializing twice? According to the docs the shape of the coefficients should be `(1, 1, 6)` did that not work?

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks for the effort. Only had a tiny question

File: tests/test_filter_classes.py (hunk 1 above)
Comment (tluebeck) [reply tests/test_filter_classes.py:277]: It was a copy and paste from the test for the non complex data, to be honest :) ... I removed. Also in the non complex test. Should I also test it with coefficients (1, 1, 6)?

Comment (f-brinkmann) [reply tests/test_filter_classes.py:277]: I tried it and filt.coefficients have shape `(1,1,6)` in both cases. but maybe inputting the coefficients according to the docs is less confusing, i.e., using `sos = np.array([[[1, 1/2, 0, 1, 0, 0]]])`

**Review by f-brinkmann — APPROVED** [review]

File: tests/test_filter.py [hunk 2]
```diff
@@ -79,6 +79,7 @@ def test_chebyshev2(impulse):
         x = pfilt.chebyshev2(None, 2, 40, 1000, 'lowpass')
 
 
+
```
Comment (ahms5) [inline tests/test_filter.py:82]:
```suggestion
```

**Review by ahms5 — CHANGES_REQUESTED** [review]
thank you for implementing

**Pull request merged**